ml_model/model/results/compare_results.py

```diff
@@ -0,0 +1,172 @@
+            writer.writerow(header)
+
+
+## draw histogram - to check the normality for diff_f1, diff_precision, diff_recall
+columns = ['diff_precision','diff_recall','diff_f1', 'diff_auc']
+for column in columns:
+    hist = overall_df.hist(column=column)
+    plt.savefig(ROOT_PATH + 'results/histograms/final_hist_' + column +'_'+ str(SMOTE_TAG) + '.png')
+
+## draw histogram for all the input features in a single plot
+df_raw_features = pd.read_csv(ROOT_PATH + 'data/understandability_with_warnings.csv')
+with open(ROOT_PATH + "classification/experiments.jsonl") as jsonl_file:
+    experiments = [json.loads(jline) for jline in jsonl_file.read().splitlines()]
+    for experiment in experiments:
+        full_dataset = df_raw_features.dropna(subset=experiment["target"])
+        feature_X_c = full_dataset[experiment["features"]].iloc[:, 1:]
+        ## draw histogram with labels
+        names = list(feature_X_c.columns)
+        feature_X_c.hist(figsize=(10, 6), bins=len(feature_X_c.columns),  stacked=True)
+        plt.suptitle('')
+        plt.tight_layout()
+        plt.savefig(ROOT_PATH + 'results/histograms/final_hist_code_features'+ experiment["target"] +'_'+ str(SMOTE_TAG) + '.png')
+        plt.close()
+
+#######################
+## STATISTICAL TESTS ##
+#######################
+tests = {'metric':'', 'model':'', 'target':'', 'wilcoxon_test(p-value)':''}
+## write header
+with open(ROOT_PATH + 'results/final_statistics_'+ str(SMOTE_TAG) + '.csv', "w") as csv_file:
+    writer = csv.DictWriter(csv_file, fieldnames=tests.keys())
+    writer.writeheader()
+
+metrics = {'diff_f1': ['f1_c','f1_cw'], 
+           'diff_precision': ['precision_c','precision_cw'], 
+           'diff_recall': ['recall_c','recall_cw'],
+           'diff_auc': ['auc_c','auc_cw']}
+
+model_names = ["SVC", "knn_classifier", "logistic_regression", "randomForest_classifier", "mlp_classifier"]
+targets = ['ABU50', 'BD50', 'PBU']
+
+for model_name in models:
+    for target in targets:
+        model_data = overall_df.query("model == '" + model_name + "' and target == '" + target + "'")
+        print (model_data.shape[0])
+        for metric in metrics.keys():
+            x = model_data[metrics[metric][0]] ## code
+            y = model_data[metrics[metric][1]] ## code+warnings
+
+            ## 1. WILCOXON TEST ##
+            ## null hypothesis: There is no statistically significant difference between x and y
+            ## if p-value <= 0.05, we reject the null hypothesis. i.e. there is a statistically significant difference between x and y
+            ## Note: to handle ties (x=y), we use the zsplit method
+            _, wilcoxon_p = wilcoxon(x, y, alternative='greater', zero_method='zsplit')
+
+            ## write results to csv
+            tests['metric'] = metric
+            tests['model'] = model_name
+            tests['target'] = target
+            tests['wilcoxon_test(p-value)'] = wilcoxon_p
```

Comment:
add the number of data points

Reply:
add also the output of running the Cliff Delta's method:
https://github.com/neilernst/cliffsDelta